url8012c8d031_dmbb966_BBMS_tgz-pJ8-testbed_JNEATtestJ8/src/gui/GUIBasicInfo.java

```diff
@@ -0,0 +1,127 @@
+package gui;
+
+import hex.Hex;
+
+import java.awt.Graphics;
+
+import javax.swing.JPanel;
+
+import clock.Clock;
+import clock.ClockControl;
+import bbms.GlobalFuncs;
+
+@SuppressWarnings("serial")
+public class GUIBasicInfo extends JPanel {
+	
+	static String hexCoords = "";
+	static String terrainType = "";
+	static String elev = "";
+	static String obsc = "";
+	static String dens = "";
+	static String obsH = "";
+	static String spots = "";
+	
+	static String unitCallsign = "";
+	static String unitType = ""; 	
+	static String unitSubLoc = "";
+	
+	public static int mouseX = 0;
+	public static int mouseY = 0;
+	public static int uniMouseX = 0;
+	public static int uniMouseY = 0;
+	
+	private final int start = 25;
+	private final int spacing = 15;
+	
+	public static Hex lastHex = null;
+	
+	public GUIBasicInfo() {
+		
+	}
+	
+	/**
+	 * Updates the basic hex info display with info for hex (x, y)
+	 */
+	public static void UpdateHexInfo(int x, int y) {
+		
+		lastHex = GlobalFuncs.scenMap.getHex(x, y);
+		if (lastHex != null) {
+			hexCoords = "Hex: (" + x + ", " + y + ")";						
+			terrainType = lastHex.tType.displayType();						
+			elev = "Elev: " + lastHex.elevation + "m";																							
+			obsc = "Obsc: " + lastHex.obscuration;			
+			dens = "Density: " + lastHex.density;						
+			obsH = "ObsH: " + lastHex.obsHeight + "m";
+			spots = "Spot: " + lastHex.numSpots;
+			
+			if (lastHex.HexUnit != null) {
+				unitCallsign = lastHex.HexUnit.callsign;
+				unitType = lastHex.HexUnit.type;
+				unitSubLoc = lastHex.HexUnit.DispSubHexMovement();
+			}
+			else {
+				unitCallsign = "";
+				unitType = "";
+				unitSubLoc = "";
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 57:66, 71:82.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.